### PR TITLE
 [DR-3430] [CVE-2022-225857] Pin snakeyaml to 1.33

### PR DESCRIPTION
**Pull request by okotsopoulos**
https://broadworkbench.atlassian.net/browse/DR-3430

This PR pins Spring Boot transitive dependency snakeyaml to 1.33 to address https://nvd.nist.gov/vuln/detail/CVE-2022-25857 (fixed in 1.31).

Bumping the pin any higher to 2.0 prevents TDR from running:
```
Caused by: java.lang.NoSuchMethodError: org.yaml.snakeyaml.constructor.SafeConstructor: method 'void <init>()' not found
        at liquibase.parser.core.yaml.YamlChangeLogParser.parse(YamlChangeLogParser.java:23)
        at liquibase.changelog.DatabaseChangeLog.include(DatabaseChangeLog.java:587)
        at liquibase.changelog.DatabaseChangeLog.handleChildNode(DatabaseChangeLog.java:363)
        at liquibase.changelog.DatabaseChangeLog.load(DatabaseChangeLog.java:318)
        at liquibase.parser.core.xml.AbstractChangeLogParser.parse(AbstractChangeLogParser.java:23)
        at liquibase.Liquibase.getDatabaseChangeLog(Liquibase.java:369)
```

We should reevaluate this pin after upgrading TDR to Spring Boot 3.

While here, I removed a pin of json-smart that was no longer needed, following our recent upgrade to Spring Boot 2.7.18 :tada:

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -174,10 +174,9 @@ configurations {
     runtimeClasspath
 }
 
-// Fix for CVE-2023-1370 which identifies a vulnerability in json-smart 2.4.8
-// 2.4.9 fixes the vulnerability and 2.4.10 fixes an additional bug
-// This override can be removed when we upgrade to spring-boot v2.6.15+, v2.7.10+, or v3.0.5+
-ext['json-smart.version'] = '2.4.10'
```
Comment (okotsopoulos) [inline build.gradle:180]: Verified:

```
(base) okotsopo@wm111-e35 jade-data-repo % ./gradlew dependencyInsight --dependency json-smart

> Task :dependencyInsight
net.minidev:json-smart:2.4.11 (selected by rule)
   variant "compile" [
      org.gradle.status              = release (not requested)
      org.gradle.usage               = java-api
      org.gradle.libraryelements     = jar (compatible with: classes)
      org.gradle.category            = library

      Requested attributes not found in the selected variant:
         org.gradle.dependency.bundling = external
         org.gradle.jvm.environment     = standard-jvm
         org.gradle.jvm.version         = 17
   ]

net.minidev:json-smart:2.4.10 -> 2.4.11
\--- com.microsoft.azure:msal4j:1.13.8
     +--- com.azure:azure-identity:1.9.1
     |    \--- compileClasspath
     \--- com.microsoft.azure:msal4j-persistence-extension:1.2.0 (requested com.microsoft.azure:msal4j:1.4.0)
          \--- com.azure:azure-identity:1.9.1 (*)

net.minidev:json-smart:[1.3.3,2.4.10] -> 2.4.11
\--- com.nimbusds:oauth2-oidc-sdk:10.7.1
     \--- com.microsoft.azure:msal4j:1.13.8
          +--- com.azure:azure-identity:1.9.1
          |    \--- compileClasspath
          \--- com.microsoft.azure:msal4j-persistence-extension:1.2.0 (requested com.microsoft.azure:msal4j:1.4.0)
               \--- com.azure:azure-identity:1.9.1 (*)

(*) - dependencies omitted (listed previously)

A web-based, searchable dependency report is available by adding the --scan option.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 8.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

See https://docs.gradle.org/7.3/userguide/command_line_interface.html#sec:command_line_warnings

BUILD SUCCESSFUL in 5s
```

File: build.gradle [hunk 2]
```diff
@@ -174,10 +174,9 @@ configurations {
...
+// Fix for CVE-2022-25857 which identifies a vulnerability in snake-yaml 1.30
+// This override can likely be removed or bumped to 2.0 when we upgrade to spring boot 3
+ext['snakeyaml.version'] = '1.33'
```
Comment (okotsopoulos) [inline build.gradle:179]: Verified:
```
(base) okotsopo@wm111-e35 jade-data-repo % ./gradlew dependencyInsight --dependency snakeyaml 

> Task :dependencyInsight
org.yaml:snakeyaml:1.33 (selected by rule)
   variant "compile" [
      org.gradle.status              = release (not requested)
      org.gradle.usage               = java-api
      org.gradle.libraryelements     = jar (compatible with: classes)
      org.gradle.category            = library

      Requested attributes not found in the selected variant:
         org.gradle.dependency.bundling = external
         org.gradle.jvm.environment     = standard-jvm
         org.gradle.jvm.version         = 17
   ]

org.yaml:snakeyaml:1.23 -> 1.33
\--- org.liquibase:liquibase-core:3.8.0
     \--- compileClasspath

org.yaml:snakeyaml:1.30 -> 1.33
\--- org.springframework.boot:spring-boot-starter:2.7.18
     +--- org.springframework.cloud:spring-cloud-gcp-starter:1.2.8.RELEASE (requested org.springframework.boot:spring-boot-starter:2.3.10.RELEASE)
     |    \--- org.springframework.cloud:spring-cloud-gcp-starter-logging:1.2.8.RELEASE
     |         \--- compileClasspath
     +--- org.springframework.boot:spring-boot-starter-web:2.7.18
     |    +--- compileClasspath (requested org.springframework.boot:spring-boot-starter-web)
     |    \--- io.opencensus:opencensus-contrib-spring:0.31.1 (requested org.springframework.boot:spring-boot-starter-web:2.1.5.RELEASE)
     |         \--- bio.terra:terra-common-lib:0.0.89-SNAPSHOT:20230524.210856-1
     |              \--- compileClasspath
     +--- org.springframework.boot:spring-boot-starter-thymeleaf:2.7.18
     |    \--- compileClasspath (requested org.springframework.boot:spring-boot-starter-thymeleaf)
     +--- org.springframework.boot:spring-boot-starter-actuator:2.7.18
     |    \--- compileClasspath (requested org.springframework.boot:spring-boot-starter-actuator)
     +--- org.springframework.boot:spring-boot-starter-validation:2.7.18
     |    \--- compileClasspath (requested org.springframework.boot:spring-boot-starter-validation)
     +--- org.springframework.boot:spring-boot-starter-json:2.7.18
     |    +--- org.springframework.boot:spring-boot-starter-web:2.7.18 (*)
     |    \--- bio.terra:externalcreds-client-resttemplate:0.72.0-SNAPSHOT:20220318.201049-1 (requested org.springframework.boot:spring-boot-starter-json:2.6.2)
     |         \--- compileClasspath
     +--- org.springframework.cloud:spring-cloud-starter:3.1.8 (requested org.springframework.boot:spring-boot-starter:2.6.15)
     |    \--- org.springframework.cloud:spring-cloud-starter-sleuth:3.1.10
     |         \--- compileClasspath
     \--- org.springframework.boot:spring-boot-starter-aop:2.7.18
          \--- org.springframework.cloud:spring-cloud-starter-sleuth:3.1.10 (requested org.springframework.boot:spring-boot-starter-aop:2.6.15) (*)

org.yaml:snakeyaml:2.0 -> 1.33
\--- io.kubernetes:client-java:18.0.0
     \--- compileClasspath

(*) - dependencies omitted (listed previously)

A web-based, searchable dependency report is available by adding the --scan option.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 8.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

See https://docs.gradle.org/7.3/userguide/command_line_interface.html#sec:command_line_warnings

BUILD SUCCESSFUL in 2s
```

**Review by snf2ye — APPROVED** [review]
Thank you!

**Review by pshapiro4broad — APPROVED** [review]
This looks OK to fix the problem, but there are newer versions of `liquibase-core` without this CVE (the version we use, 3.8.0, is from 2019).  It looks like that may be updated as part of https://broadworkbench.atlassian.net/browse/DR-3420, so we should be sure to try and remove the pin then.

**Comment by snf2ye** [comment]
> This looks OK to fix the problem, but there are newer versions of `liquibase-core` without this CVE (the version we use, 3.8.0, is from 2019). It looks like that may be updated as part of https://broadworkbench.atlassian.net/browse/DR-3420, so we should be sure to try and remove the pin then.

Oh, ok - I'll see if my liquibase-core upgrade fixes this issue.

Update: My PR upgrades liquibase-core from 3.8.0 to 4.9.1 by unpinning the version. But, it appears the snakeyaml version is still 1.30.

**Comment by okotsopoulos** [comment]
> Oh, ok - I'll see if my liquibase-core upgrade fixes this issue.
>
> Update: My PR upgrades liquibase-core from 3.8.0 to 4.9.1 by unpinning the version. But, it appears the snakeyaml version is still 1.30.

Thank you for checking!

**Pull request merged**